src/Microsoft.DotNet.Interactive/KernelExtensions.cs

```diff
@@ -174,7 +174,44 @@ public static CompositeKernel UseProxyKernelWithNamedPipe(this CompositeKernel k
 
                 }
             });
-            kernel.ConfigureConnection(connectionCommand);
+            kernel.AddConnectionDirective(connectionCommand);
+            return kernel;
+        }
+
+        public static CompositeKernel UseProxyKernelWithSignalR(this CompositeKernel kernel)
+        {
+            var connectionCommand = new Command("signalr");
+            connectionCommand.AddArgument(new Argument<string>("kernel-name"));
+            connectionCommand.AddArgument(new Argument<string>("hub-url"));
+
+            connectionCommand.Handler = CommandHandler.Create<string, string, KernelInvocationContext>(async (kernelName, hubUrl, context) =>
+            {
+                var existingProxyKernel = kernel.FindKernel(kernelName);
+                if (existingProxyKernel == null)
+                {
+                  
+                    var connection = new HubConnectionBuilder()
+                        .WithUrl(hubUrl)
+                        .Build();
+
+                    await connection.StartAsync();
+
+                    var client = connection.CreateKernelClient();
+                    var proxyKernel = new ProxyKernel(kernelName, client);
+                    await connection.SendAsync("connect");
+
+                    proxyKernel.RegisterForDisposal(client);
+                    proxyKernel.RegisterForDisposal(async () =>
```

Comment:
This looks like an async void and should be avoided. Looks like we should add `IAsyncDisposable` support to `PocketDisposable`.

Reply:
ok but that one  is using rx CompositeDisposable.

Reply:
It's an implementation detail, and since we can change the code for `PocketDisposable`, this seems pretty straightforwrd.